planner/core/task.go

```diff
@@ -1256,6 +1256,11 @@ func (p *PhysicalLimit) sinkIntoIndexLookUp(t task) bool {
 	if !isTableScan {
 		return false
 	}
+	// We can't sink Limit into IndexLookUpReader when it scans a partition table in dynamic prune mode
+	// https://github.com/pingcap/tidb/issues/32516
```

Comment:
I think it is just a quick fix? I think the root cause is that `IndexLookUpExecutor` cannot handle limit push down for partitioned table correctly. I just read the related code found `IndexLookUpExecutor` is doing limit in `indexWorker.extractTaskHandles`:

https://github.com/pingcap/tidb/blob/cc789d078b44d3ab41fd907e784c2e8acf242fad/executor/distsql.go#L907-L909

However for partitioned table, it may have more than one `indexWorker` .

Reply:
Maybe the best way is move the limit works out of indexWorker

Reply:
Maybe one question here. If we have more than one indexWorkers, how to deal with the `offset` in PushedLimit. I have no idea about it.